### PR TITLE
Wla 514

### PR DESCRIPTION
**Pull request by rishikrsharma**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)
WLA-514

DIVORCE: New CCD Event to update case with translated details
## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

**Test Configuration**:

* Hardware:
* O/S and version:
* JDK:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/divorce/documentgenerator/mapper/WelshTemplateDataMapper.java [hunk 1]
```diff
@@ -0,0 +1,629 @@
...
+    private void translatedWelshData(Map<String, Object> data) {
+
+        Map<String, String> relationship =
+                templateConfig.getTemplate().get(RELATION).get(WELSH);
+
+        Optional.ofNullable(data.get(D8_DIVORCE_WHO_KEY)).ifPresent(
+            divorceWho ->
+                    data.put(WELSH_D8_DIVORCE_WHO_KEY, relationship.get(divorceWho)));
+
+        Optional.ofNullable(data.get(D8_MARRIAGE_DATE_KEY)).map(String.class::cast).ifPresent(
```
Comment (ruban72) [inline src/main/java/uk/gov/hmcts/reform/divorce/documentgenerator/mapper/WelshTemplateDataMapper.java:201]: Refactor to create reusable method here

File: src/main/java/uk/gov/hmcts/reform/divorce/documentgenerator/mapper/WelshTemplateDataMapper.java [hunk 2]
```diff
@@ -0,0 +1,629 @@
...
+            date -> data.put(WELSH_ADULTERY_FOUND_OUT_DATE_KEY, localDateToWelshStringConverter.convert(date)));
+
+        Optional.ofNullable(data.get(DECREE_NISI_SUBMITTED_DATE_KEY)).map(String.class::cast).ifPresent(
+            date -> data.put(WELSH_DECREE_NISI_SUBMITTED_DATE_KEY, localDateToWelshStringConverter.convert(date)));
+
+        Optional<String> d8ReasonForDivorceBehaviourDetailsFlag =
+            Optional.ofNullable(data.get(D8_REASON_FOR_DIVORCE_BEHAVIOUR_DETAILS_TRANS_LANG))
+            .map(String.class::cast);
+
+        if (d8ReasonForDivorceBehaviourDetailsFlag.isPresent()) {
```
Comment (ruban72) [inline src/main/java/uk/gov/hmcts/reform/divorce/documentgenerator/mapper/WelshTemplateDataMapper.java:288]: Refactor to create reusable method here

File: src/main/java/uk/gov/hmcts/reform/divorce/documentgenerator/mapper/WelshTemplateDataMapper.java (hunk 1 above)
Comment (rishikrsharma) [reply src/main/java/uk/gov/hmcts/reform/divorce/documentgenerator/mapper/WelshTemplateDataMapper.java]: Done

File: src/main/java/uk/gov/hmcts/reform/divorce/documentgenerator/mapper/WelshTemplateDataMapper.java (hunk 2 above)
Comment (rishikrsharma) [reply src/main/java/uk/gov/hmcts/reform/divorce/documentgenerator/mapper/WelshTemplateDataMapper.java:251]: Done

File: src/main/java/uk/gov/hmcts/reform/divorce/documentgenerator/mapper/WelshTemplateDataMapper.java [hunk 3]
```diff
@@ -0,0 +1,587 @@
...
+        } else {
+            setWelshTranslatedData(data,D8_REASON_FOR_DIVORCE_BEHAVIOUR_DETAILS_EN,
+                D8_REASON_FOR_DIVORCE_BEHAVIOUR_DETAILS_CY,D8_REASON_FOR_DIVORCE_BEHAVIOUR_DETAILS,
+                D8_REASON_FOR_DIVORCE_BEHAVIOUR_DETAILS);
+        }
+
+        Optional<String> d8ReasonForDivorceAdulteryDetailsFlag =
+            getLanguagePreferenceFlag(data, D8_REASON_FOR_DIVORCE_ADULTERY_DETAILS_TRANS_LANG);
+
+        if (d8ReasonForDivorceAdulteryDetailsFlag.isPresent()) {
```
Comment (ruban72) [inline src/main/java/uk/gov/hmcts/reform/divorce/documentgenerator/mapper/WelshTemplateDataMapper.java:270]: Refactor this call out to reusable method as well please

**Pull request merged**